test/e2e/network_test.go

```diff
@@ -566,24 +597,51 @@ func (tc *testContext) deleteJob(name string) error {
 func (tc *testContext) waitUntilJobSucceeds(name string) error {
 	var job *batchv1.Job
 	var err error
+	var labelSelector string
 	for i := 0; i < retryCount; i++ {
 		job, err = tc.client.K8s.BatchV1().Jobs(tc.workloadNamespace).Get(context.TODO(), name, metav1.GetOptions{})
 		if err != nil {
 			return err
 		}
+		labelSelector = "job-name=" + job.Name
 		if job.Status.Succeeded > 0 {
+			tc.savePodLogs(labelSelector)
 			return nil
 		}
 		if job.Status.Failed > 0 {
+			tc.savePodLogs(labelSelector)
 			events, _ := tc.getPodEvents(name)
 			return errors.Errorf("job %v failed: %v", job, events)
 		}
 		time.Sleep(retryInterval)
 	}
+	tc.savePodLogs(labelSelector)
 	events, _ := tc.getPodEvents(name)
 	return errors.Errorf("job %v timed out: %v", job, events)
 }
 
+// savePodLogs prints the logs associated with the label
+// selector of a given pod job or deployment to the ARTIFACT_DIR
+func (tc *testContext) savePodLogs(labelSelector string) {
+	logs, err := tc.getLogs(labelSelector)
+	if err != nil {
+		log.Printf("Unable to get logs associated with pod: %s", labelSelector)
+		return
+	}
+	podLogFile := "pod-" + labelSelector + ".log"
+	podArtifacts := filepath.Join(os.Getenv("ARTIFACT_DIR"), "pods")
+	podDir := filepath.Join(podArtifacts, labelSelector)
+	err = os.MkdirAll(podDir, os.ModePerm)
+	if err != nil {
+		log.Printf("Error creating pod log collection directory")
+	}
+	outputFile := filepath.Join(podDir, filepath.Base(podLogFile))
+	logsErr := ioutil.WriteFile(outputFile, []byte(logs), os.ModePerm)
+	if logsErr != nil {
+		log.Printf("Unable to print pod logs with label %s to a file", labelSelector)
```

Comment:
Consider adding the file name in the error message

```suggestion
		log.Printf("Unable to print pod logs with label %s to file %s", labelSelector, <filename>)
```